src/pytest_subtests/plugin.py

```diff
@@ -98,6 +100,24 @@ def _from_test_report(cls, test_report: TestReport) -> SubTestReport:
         return super()._from_json(test_report._to_json())
 
 
+def _addSkip(self: TestCaseFunction, testcase: TestCase, reason: str) -> None:
+    if isinstance(testcase, _SubTest):
+        self._originaladdSkip(testcase, reason)  # type: ignore[attr-defined]
+        if self._excinfo is not None:
+            exc_info = self._excinfo[-1]
+            self.addSubTest(testcase.test_case, testcase, exc_info)  # type: ignore[attr-defined]
+    else:
+        # The non-subtest skips have to be added by `_originaladdSkip` only after all subtest failures are processed by
+        # `_addSubTest`.
+        if (
+            len(
+                [x for x, y in self.instance._outcome.errors if isinstance(x, _SubTest)]
+            )
+            == 0
+        ):
+            self._originaladdSkip(testcase, reason)  # type: ignore[attr-defined]
```

Comment:
If we have skipTest outside a subTest and there are some subtest failures, this has to be done in this way here to make the reporting results correct.